components/Dashboard/Tiles/BarChartTile/BarChartTile.js

```diff
@@ -1,27 +1,26 @@
 import { useSession } from "next-auth/react";
 import experienceAnalyser from "@/utils/experienceAnalyser";
-import lastWeekAnalyser from "@/utils/lastWeekAnalyser";
-import { useData } from "@/lib/useData";
 import * as Styled from "./BarChartTile.styled";
 
 export default function BarChartTile({
-  isLastWeek,
   handleFilterSwitchClick,
   singleExperienceList,
+  isLast7Days,
+  setIsLast7Days,
   handleExperienceClick,
   setSingleExperienceList,
   clickedExperience,
   singleEmotionDisplayed,
+  allEntries,
+  userEntries,
+  isLoadingEntries,
+  errorEntries,
+  last7DaysEntries,
 }) {
   const { data: session } = useSession();
 
-  const { allEntries, isLoadingEntries, errorEntries } =
-    useData().fetchedAllEntries;
-  const { userEntries } = useData().fetchedUserEntries;
-  const lastWeek = lastWeekAnalyser(session ? userEntries : allEntries);
-
-  const visualizedData = isLastWeek
-    ? experienceAnalyser(lastWeek)
+  const visualizedData = isLast7Days
+    ? experienceAnalyser(last7DaysEntries && last7DaysEntries)
     : experienceAnalyser(session ? userEntries : allEntries);
```

Comment:
in this line you decide which data should be visualized. You can move this logic out of this component to Dashboard.js and only pass down a generic prop called "data" or "experiences" which can be filled with any array. Then you don't need to pass down a lot of these props in the first place (experienceAnalyzer, user Entries allEntries, etc.)
for example: 

```js
// Dashboard.js

const visualizedData = isLast7Days
    ? experienceAnalyser(last7DaysEntries && last7DaysEntries)
    : experienceAnalyser(session ? userEntries : allEntries);
const experiences = visualizedData.experiences

...

<EntriesColumn experiences={experiences}>

//EntriesColumn.js

<BarChartTile epxeriences={experiences} />

//BarChartTile.js

experiences.map(...)
```